File: src/main/java/com/baulsupp/oksocial/services/google/GoogleUtil.java
package com.baulsupp.oksocial.services.google;

import com.google.common.collect.Sets;
import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.Set;

public class GoogleUtil {
  public static final Collection<String> SCOPES =
      Arrays.asList("plus.login", "plus.profile.emails.read");

  private GoogleUtil() {
  }

  public static final Set<String> API_HOSTS =
      Collections.unmodifiableSet(Sets.newHashSet(
          "api.google.com")
      );

  public static String fullScope(String suffix) {
    return suffix.contains("/") ? suffix : "https://www.googleapis.com/auth/" + suffix;
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/services/google/GoogleUtil.kt
package com.baulsupp.oksocial.services.google

object GoogleUtil {
  val SCOPES = listOf("plus.login", "plus.profile.emails.read")

  val API_HOSTS = setOf((
    "api.google.com")
  )

  fun fullScope(suffix: String): String {
    return if (suffix.contains("/")) suffix else "https://www.googleapis.com/auth/$suffix"
  }
}
